### ISSUE TITLE
Pylint cannot find `misc_util` module in `numpy.distutils`

### ISSUE THREAD

**Issue by Thijss** [PyCQA/pylint#6497]
### Bug description

Hi there,

No sure whether this is an issue in numpy or pylint.

Starting from `pylint` 2.13.0, `pylint` reports an error when importing `numpy.distutils.misc_util`
When I downgrade pylint to the previous version (2.12.2) there is no error reported

script `a.py`:
```
# pylint: disable=unused-import,missing-module-docstring

from numpy.distutils.misc_util import is_sequence


assert is_sequence("ABC") is False
assert is_sequence([1, 2, 3]) is True

```

### Command used

```shell
pylint a.py
```

### Pylint output

```shell
************* Module a
a.py:3:0: E0611: No name 'misc_util' in module 'numpy.distutils' (no-name-in-module)
a.py:3:0: E0401: Unable to import 'numpy.distutils.misc_util' (import-error)
```

### Expected behavior

The module exists so I would not expect pylint to report an error here 

### Pylint version

```shell
2.13.0
```

### OS / Environment

Mac OS 12.3.1

### Additional dependencies

numpy==1.22.3

**Comment by jacobtylerwalls** [comment]
Thanks for the report. I bisected it to https://github.com/PyCQA/astroid/pull/1386. Apparently `numpy` has its own `distutils` module, and we need a smarter to test to distinguish `virtualenv`'s patching of `distutils` from other packages like `numpy` that have their own versions.

**Comment by oh-my-lawdy** [comment]
I'm seeing a similar issue, but with `pydantic` instead of `numpy`.

a.py
```
from pydantic import BaseModel

```

**Command Used**
```
pylint a.py
```

**Pylint Output**
```
************* Module test
test.py:1:0: C0114: Missing module docstring (missing-module-docstring)
test.py:1:0: E0611: No name 'BaseModel' in module 'pydantic' (no-name-in-module)
test.py:1:0: W0611: Unused BaseModel imported from pydantic (unused-import)

------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 7.50/10, -7.50)
```

**Pylint Version**
```
2.13.8
```

**OS/Environment**
```
Amazon Linux 2
```

**Other Dependencies**
```
pydantic==1.9.0
```
